### PR TITLE
[HTP-115] Kargo support for IAB US privacy string

### PR DESCRIPTION
**Pull request by samuelhorwitz**
## Type of Change

- [ ] New adapter
- [x] Feature Update
- [ ] Bug Fix
- [ ] Code Refactor
- [ ] Other

## Description of Change
Adding CCPA support

### CODE REVIEW

**Comment by samuelhorwitz** [comment]
When I test this locally, it does not appear that `usp` on `ComplianceService` is defined. I have branched off canary. Can we assume that this will be defined always and it just hasn't been yet or should I add safety checks?

**Comment by jas-singh** [comment]
@samuelhorwitz 

Yes,  `usp` property will always be available if publisher has enabled IX library's privacy features (which can be determined by querying` ComplianceService.isPrivacyEnabled()` in the adapter). Additionally, the `uspString` property of the object retrieved from `ComplianceService.usp.getConsent()` will always exist as well. Please note the `uspString` property will be set as an empty string in the scenarios where the IX Library is not able to retrieve consent/opt-out from the site's page

**Comment by samuelhorwitz** [comment]
Thank you! I'll add a safety check then just to be sure.

File: kargo/kargo-htb.js [hunk 1]
```diff
@@ -165,13 +166,16 @@ function KargoHtb(configs) {
...
+        var uspConsentObj = ComplianceService.usp && ComplianceService.usp.getConsent();
 
         return {
             kargoID: crb.userId || '',
             clientID: crb.clientId || '',
             tdID: __getTDID(returnParcels),
             crbIDs: crb.syncIds || {},
-            optOut: crb.optOut || false
+            optOut: crb.optOut || false,
+            usp: privacyEnabled && uspConsentObj ? uspConsentObj.uspString : null
```
Comment (ix-certification) [inline kargo/kargo-htb.js]: @samuelhorwitz

When the IX Library is not able to retrieve CCPA information the `ComplianceService.usp.getConsent()` function will return an object with the `uspString` property set to an empty string (ex. `{version: 1, uspString: ""}`).

In that case, can you confirm what the expected value for `usp` will need to be (null or empty string)?

If `usp` value is expected to be `null` then the value of `uspConsentObj.uspString ` will need to be checked before assigning it to the `usp` property.

Comment (samuelhorwitz) [reply kargo/kargo-htb.js]: Our backend will handle null and empty string equivalently for the IAB string

Comment (ix-certification) [reply kargo/kargo-htb.js]: Perfect. In that case the changes look good and we will move forward with merging into master.

**Pull request merged**